engine/resources/exec.go

```diff
@@ -138,9 +141,38 @@ func (obj *ExecRes) Validate() error {
 		}
 	}
 
+	// check that environment variables' format is valid
+	for key := range obj.Env {
+		if !isNameValid(key) {
+			return fmt.Errorf("environment variable name must only contain alphanumeric characters or underscore")
+		}
+	}
+
 	return nil
 }
 
+// TODO: move this function further down in code
+// checks that environment variable name is valid
+func isNameValid(varName string) bool {
```

Comment:
change sig to return `error` instead. Instead of `return false` use fmt.Errorf(...reason...) instead.